### PR TITLE
Minor bug fixes

### PR DESCRIPTION
**Pull request by p3g4asus**
Hi,
with this PR I am fixing some minor bug I found in the code. They are:
**Keep last value of progress and ETA when parsing fails
Fix cache option presence checking
When dumping json of a playlist, do not throw exception if the playlist has only some "erratic" videos and not all**
Thanks for your work.

### CODE REVIEW

**Review by yausername — APPROVED** [review]
Left a comment, but I'm approving the changes.
Thanks for the contribution!

File: library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.java [hunk 1]
```diff
@@ -196,8 +196,9 @@ public YoutubeDLResponse execute(YoutubeDLRequest request, @Nullable DownloadPro
 
         String out = outBuffer.toString();
         String err = errBuffer.toString();
+        boolean json = request.hasOption("--dump-json");
 
-        if (exitCode > 0) {
+        if (exitCode > 0 && (!json || (json && out.isEmpty()))) {
```
[the remark this answers is not in the document]
Comment (p3g4asus) [reply library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.java:201]: I see. I have to make some tests. About throwing an exception on invalid JSON, you are definitely right. I will try to make this part more robust.

[the remark this answers is not in the document]
Comment (p3g4asus) [reply library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.java:201]: I have mede one more test. `--ignore-errors` does not prevent ytdlp returning 1 and not 0 when having a single erratic video in a playlist.
We could change the condition as follows
```java
        if (exitCode > 0 && (!json || (json && (out.isEmpty() || !request.hasOption("--ignore-errors"))))) {
```
For json sanity checking we could use the code inside `getInfo`. Obviously we should try to parse the output only if the above condition fails and `exitCode` is > 0. What do you think?

**Comment by p3g4asus** [comment]
Now it should be OK.

**Comment by yausername** [comment]
Looks good. Feel free to merge.

**Pull request merged**